### ISSUE TITLE
Error loading module created by MSVC

### ISSUE THREAD

**Issue by houghtonap** [sqlitebrowser/sqlitebrowser#2143]
Details for the issue
--------------------

1. Pulled the source code for the csv extension module csv.c from
    https://sqlite.org/src/artifact/3ed979c1eb35e35a
1. Placed the sqlite3.h and sqlite3ext.h from amalgamation zip file in current directory:
    https://www.sqlite.org/2020/sqlite-amalgamation-3310100.zip
1. Compiled csv.c with MSVC command: 
    cl -I . csv.c -link -dll -out:csv.msvc.dll
1. Opened DB Browser for SQLite.
1. Clicked on New Database button.
1. Clicked on Cancel button.
1. Clicked on Tools -> Load Extension... menu choice.
1. Selected csv.msvc.dll and clicked Open button on Select extension file dialog box.

#### What did you expect to see?

1. Expected the message Extension successfully loaded on the DB Browser for SQLite dialog box.
1. Expected to execute the following SQL and it work:
    drop table if exists temp.csv ;
    create virtual table temp.csv using csv
    (
        filename = "u:\Documents\Source\sqlite\Test.csv",
        header = 1
    ) ;
    select * from csv ;
1. Extension loads in SQLiteStudio and the above SQL works as expected.

#### What did you see instead?

1. Dialog box appeared with:
    Error loading extension: The specified module could not be found.

Workaround tried but failed
---------------------------

1. Pulled the source code for the csv extension module csv.c from
    https://sqlite.org/src/artifact/3ed979c1eb35e35a
1. Opened msys MinGW 64 shell.
1. Loaded msys package https://packages.msys2.org/base/mingw-w64-sqlite3
    pacman -S mingw-w64-x86_64-sqlite3
1. Compiled csv.c under msys2 using mingw64 gcc:
    gcc -g -shared -I/mingw64/include csv.c -o csv.mingw64.dll
1. Opened DB Browser for SQLite.
1. Clicked on New Database button.
1. Clicked on Cancel button.
1. Clicked on Tools -> Load Extension... menu choice.
1. Selected csv.mingw64.dll and clicked Open button on Select extension file dialog box.
1. Received Extension successfully loaded on the DB Browser for SQLite dialog box.
1. Ran the following SQL and DB Browser crashed:
    drop table if exists temp.csv ;
    create virtual table temp.csv using csv
    (
        filename = "u:\Documents\Source\sqlite\Test.csv",
        header = 1
    ) ;
    select * from csv ;
1. Contents of Test.csv
    id,text
    1,hello world
    2,me me
    3,abc def
1. Expected select statement to generate a two column table ( id, text ) with 3 rows.

Useful extra information
-------------------------

> Version 3.11.2
> Built for x86_64-little_endian-llp64, running on x86_64
> Qt Version 5.11.3
> SQLite Version 3.27.2

**Comment by karim** [comment]
I did all the steps you mentioned to build the csv extension using MSVC and it's working for me.

![2143](https://user-images.githubusercontent.com/8761081/75077576-e3055480-550b-11ea-94a3-a50fced20fa1.png)

Did you use **x64 Native Tools Command Prompt** to build the dll?

**Comment by houghtonap** [comment]
Yes, I believe I did, but Monday I'll redo again to double check since it worked for you.

________________________________

**Comment by houghtonap** [comment]
Forgot to update this... The extension was compiled with x32 on an x64 computer because the wrong shell environment was brought up, oops. However, I still think that if someone accidentally does this, that it should not crash the application. Perhaps some additional error handling might avoid the application crash. Not sure whether this should be closed, relabeled a bug or enhancement to add the additional error handling.

**Comment by chrisjlocke** [comment]
I'll leave this open with a view to add a bit more error handling - the application shouldn't crash, but I have no idea if the error can be trapped at all, or whether as the module has been kicked in the goolies, DB4S just doesn't stand a chance of standing there.
